code_map.py

```diff
@@ -1,296 +1,346 @@
+        if not path.isfile(pack):
+            pack = path.join(ipath(), 'sublime-codemap.sublime-package')
 
         zip = zipfile.ZipFile(pack)
-        
-        for syntax in default_mappers: 
-            if not os.path.isfile(mapper_path(syntax)): 
+
+        for syntax in default_mappers:
+            if not path.isfile(mapper_path(syntax)):
                 zip.extract('custom_mappers/code_map.'+syntax+'.py', dst)
 
-        for syntax in custom_languages: 
-            if not os.path.isfile(syntax_path(syntax)): 
+        for syntax in custom_languages:
+            if not path.isfile(syntax_path(syntax)):
                 zip.extract('custom_languages/'+syntax+'.sublime-syntax', dst)
 
-    else:  
+    else:
         # package was installed manually
-        plugin_dir = os.path.dirname(__file__)
+        plugin_dir = path.dirname(__file__)
 
-        for syntax in default_mappers: 
-            src_mapper = os.path.join(plugin_dir, 'custom_mappers', 'code_map.'+syntax+'.py')
+        for syntax in default_mappers:
+            src_mapper = path.join(
+                plugin_dir, 'custom_mappers', 'code_map.'+syntax+'.py')
             dst_mapper = mapper_path(syntax)
-            if not os.path.isfile(dst_mapper): 
-                shutil.copyfile(src_mapper, dst_mapper) 
+            if not path.isfile(dst_mapper):
+                shutil.copyfile(src_mapper, dst_mapper)
 
-        for syntax in custom_languages: 
-            src_syntax = os.path.join(plugin_dir, 'custom_languages', syntax+'.sublime-syntax')
+        for syntax in custom_languages:
+            src_syntax = path.join(
+                plugin_dir, 'custom_languages', syntax+'.sublime-syntax')
             dst_syntax = syntax_path(syntax)
-            if not os.path.isfile(syntax): 
-                shutil.copyfile(src_syntax, dst_syntax) 
+            if not path.isfile(syntax):
+                shutil.copyfile(src_syntax, dst_syntax)
 
     # make a list of the available mappers
     MAPPERS = os.listdir(mpdir)
-# -------------------------
-def settings():
-    return sublime.load_settings("code_map.sublime-settings")
-# -------------------------
-def code_map_file():
 
-    plugin_dir = ''
+# -------------------------
 
-    if hasattr(sublime, 'cache_path'):
-        plugin_dir = sublime.cache_path()
-    else:
-        plugin_dir = 'cache'
-        plugin_dir = os.path.join(os.getcwd(), plugin_dir)
 
-    data_dir = path.join(plugin_dir, 'CodeMap', 'CodeMap.Data', str(sublime.active_window().id()))
-    if not path.exists(data_dir):
-        os.makedirs(data_dir)
-    return path.join(data_dir, 'Code - Map')
-# -----------------
-def set_layout_columns(count, coll_width=0.75):
+def code_map_file():
 
-    if count == 1:
-        sublime.active_window().run_command("set_layout", {"cells": [[0, 0, 1, 1]], "cols": [0.0, 1.0], "rows": [0.0, 1.0]})
+    dst = path.join(packages_dir, 'User', 'CodeMap')
+    return path.join(dst, 'Code - Map')
 
```

Comment:
no cache, there's the user folder for user files, and it's the same for all installations, also portable